src/provider/view/components/AppCard.tsx

```diff
@@ -23,10 +23,11 @@ export function AppCard(props: AppCardProps): React.ReactElement {
         event.stopPropagation();
         openHandler(app);
     };
+    const className = 'app-card w3-card w3-round-large w3-button' + (selected ? ' selected' : '');
     return (
-        <div className={'app-card w3-card w3-round-large w3-button' + (selected ? ' selected' : '')} onClick={clickHandler} onDoubleClick={doubleClickHandler}>
+        <div id={app.name} className={className} onClick={clickHandler} onDoubleClick={doubleClickHandler}>
```

Comment:
Is there any reason for adding an `id` attribute? Also can `app.name` have spaces (invalid css/html selector characters) in it?

Reply:
This is for the integration tests that need to fake clicks on specific buttons. Given that react escapes id strings so that you can't do anything actually breaking, I feel like I wouldn't worry too much about spaces (also, it all seems to work fine even with spaces in ids)

Reply:
Good point here. I'm not sure about the format of `name` (nor `appId`); in the two sample directories we have (for demo app and the tests), all `name`s happen to be valid html IDs, but for example some `appId`s on the test directory (should we decide at some point to go with those) currently aren't.
Perhaps it wouldn't hurt to add some escaping logic on both the resolver and the tests for extra robustness.